src/cargo/ops/cargo_new.rs

```diff
@@ -258,10 +258,47 @@ fn check_name(
             name
         ))?;
     }
+    if name != to_snake_case(name) {
+        shell.warn(format!(
+            "the name `{}` does not follow the preferred snake case format\n\
+            See https://github.com/rust-lang/rfcs/blob/master/text/0430-finalizing-naming-conventions.md#General-naming-conventions",
+            name
+        ))?;
+    }
 
     Ok(())
 }
 
+fn to_snake_case(mut str: &str) -> String {
```

Comment:
Why not lowercase the whole string at once? It seems you assume significance of the upper case characters but is that worthwhile for this functionality? https://devdocs.io/rust/std/primitive.str#method.to_lowercase

Reply:
Wanted to be in sync with what the compiler did. But I see your point.  Will update.